vernac/comFixpoint.ml

```diff
@@ -261,7 +261,7 @@ let declare_fixpoint local poly ((fixnames,fixdefs,fixtypes),pl,ctx,fiximps) ind
         fixdefs) in
     let evd = Evd.from_ctx ctx in
     Lemmas.start_proof_with_initialization (local,poly,DefinitionBody Fixpoint)
-      evd pl (Some(false,indexes,init_tac)) thms None (Lemmas.mk_hook (fun _ _ -> ()))
+      evd pl (Some(false,indexes,init_tac)) thms None (Lemmas.mk_hook (fun _ _ -> Global.set_typing_flags original_typing_flag))
```

Comment:
This seems quite fragile. Let's try to come up with a better model for how to do this kind of thing.

Reply:
Maybe it should be handled more like how Program Mode is done.
OTOH
~~~coq
Require Import Program.Tactics.
Program Lemma foo : Type.
Proof.
  Test Program Mode. (* off *)
~~~
so that doesn't quite work.

In general attributes which act by setting a global option seem difficult to get right especially with proof-starting commands like Lemma.

Does it even make sense to have these typing flags in the global environment? Maybe we should pass them through the constant_entry/mind_entry/etc.This would mean handling them more like universe polymorphism, with an option which is used only to get the default value of the attribute. start_proof takes a goal_kind, if we put typing flags in there we may be able to make it work in the proof too.

Reply:
Do other coq devs have any opinion?
Is it blocking or is it an "improvement request"?
Is `goal_kind` the right place for typing flags?

Reply:
The right place will be in `Declare.CInfo.t` once #12372 is finished.